### PR TITLE
fix(CSS): add :focus vs :focus-visible section

### LINKED ISSUE #27993: The explanation of the :focus-visible psuedo-class is not clear about (text) input
### MDN URL

https://developer.mozilla.org/en-US/docs/Web/CSS/:focus-visible

### What specific section or headline is this issue about?

_No response_

### What information was incorrect, unhelpful, or incomplete?

The :focus-visible docs are written in a way that make it seem that there is a way to distinguish between :focus and :focus-visible for (text) inputs. However, both these pseudo-classes seem to do the same thing for inputs.

Only when clicking in the interactive [Basic Example](https://developer.mozilla.org/en-US/docs/Web/CSS/:focus-visible#basic_example) it became clear that there is only a distinction between the two pseudo-classes for buttons and **not** for input fields.

### What did you expect to see?

A clear explanation that the pseudo-class does **not** affect input fields, but only other interactive elements.

### Do you have any supporting links, references, or citations?

_No response_

### Do you have anything more you want to share?

All other information about :focus-visible is perfectly clear and the interactive examples help a lot!

### PR DESCRIPTION
**Pull request by OnkarRuikar**
- Fixes https://github.com/mdn/content/issues/27993

### CODE REVIEW

File: files/en-us/web/css/_colon_focus-visible/index.md [hunk 1]
```diff
@@ -21,6 +21,12 @@ This selector is useful to provide a different focus indicator based on the user
 }
 ```
 
+## :focus vs :focus-visible
+
+The default behavior of user agents is to indicate focus on an element only when it is most useful to the user. For instance, when a button is clicked using a pointing device, the focus is not visually indicated. However, when interacting with an input box that requires keyboard input, the focus is indicated. When navigating using a keyboard, focus is always indicated, even on elements such as buttons.
```
Comment (estelle) [inline files/en-us/web/css/_colon_focus-visible/index.md:26]:
```suggestion
While focus styles are required when users are navigating the page with the keyboard or when focus is managed via scripts, focus styles are not required on all focusable elements when the user is using a pointing device such as a mouse or finger. For instance, when a button is clicked using a pointing device, the focus is generally not visually indicated, but when a text box needing user input has focus, focus is indicated.

The `:focus` pseudo-class always matches the currently-focused element. The `:focus-visible` pseudo-class also matches the focused element if the user needs to be informed where the focus currently is. Because the `:focus-visible` pseudo-class matches the focused element when needed, using the `:focus-visible` (instead of the `:focus` pseudo-class) allows authors to change the appearance of the focus indicator without changing when the focus indicator appears.

Originally, user-agent CSS set focus styles based only on the `:focus` pseudo-class, styling most focused elements with a focus ring outline. This meant all elements, including all links and buttons, had a focus ring applied when focused, which many found to be ugly. Because of the appearance of the focus ring, some authors removed the user-agent outline focus styles. Changing focus style can negatively impact usability. Removing focus styles makes keyboard navigation inaccessible for sighted users.
```

I may have gone a little too wordy, but wanted to make sure it's really clear. generally, I am not into "history", but here I think it helps explain things.

File: files/en-us/web/css/_colon_focus-visible/index.md [hunk 2]
```diff
@@ -21,6 +21,12 @@ This selector is useful to provide a different focus indicator based on the user
...
+
+When the [`:focus`](/en-US/docs/Web/CSS/:focus) pseudo-class is used, it always targets the currently focused element. This means that when a user employs a pointing device, a visible focus ring appears around elements like buttons, which is obtrusive. On the other hand, the `:focus-visible` pseudo-class respects the selective focus indication behavior of user agents while still allowing customization of the focus indicator.
```
Comment (estelle) [inline files/en-us/web/css/_colon_focus-visible/index.md:28]:
```suggestion
When the [`:focus`](/en-US/docs/Web/CSS/:focus) pseudo-class is used, it always targets the currently focused element. This meant that when a user employed a pointing device, a visible focus ring appeared around each focused elements, which some considered obtrusive. The `:focus-visible` pseudo-class respects user agents' selective focus indication behavior while still allowing focus indicator customization.
```
(Grammarly rewrote this for me ;) )

**Review by estelle — COMMENTED** [review]
i suggested a lot of added text. I think understanding the rationale will better help people understand the difference.

File: files/en-us/web/css/_colon_focus-visible/index.md (hunk 1 above)
Comment (OnkarRuikar) [reply files/en-us/web/css/_colon_focus-visible/index.md:26]: For better experience, shouldn't the history come first?

I wish they had mentioned this drawback of `:focus` clearly in the specs. Also other articles on Internet are not on point.

File: files/en-us/web/css/_colon_focus-visible/index.md (hunk 2 above)
Comment (OnkarRuikar) [reply files/en-us/web/css/_colon_focus-visible/index.md:28]: Looks like you've integrated grammarly with GitHub. :D

The `:focus` isn't deprecated. These things still happen.
```suggestion
When the [`:focus`](/en-US/docs/Web/CSS/:focus) pseudo-class is used, it always targets the currently focused element. This means that when a user employs a pointing device, a visible focus ring appears around the focused element, which some consider obtrusive. The `:focus-visible` pseudo-class respects user agents' selective focus indication behavior while still allowing focus indicator customization.
```

File: files/en-us/web/css/_colon_focus-visible/index.md (hunk 1 above)
Comment (estelle) [reply files/en-us/web/css/_colon_focus-visible/index.md:26]: Yes, you are right, it might make more sense for history to come first. I was thinking "if they don't understand, they're still reading, this may make it stick" but if we give them the history first, they won't be confused after three  paragraphs!

Comment (estelle) [inline files/en-us/web/css/_colon_focus-visible/index.md:26]:
```suggestion
Originally, user-agent CSS set focus styles based only on the `:focus` pseudo-class, styling most focused elements with a focus ring outline. This meant all elements, including all links and buttons, had a focus ring applied when focused, which many found ugly. Because of the appearance, some authors removed the user-agent outline focus styles. Changing focus style can negatively impact usability, while removing focus styles makes keyboard navigation inaccessible for sighted users.

Browser no longer visibly indicate focus (such as by drawing a “focus ring”), around each element when it has focus. Instead, they use a variety of heuristics to provide focus indicators only when it would be most helpful to the user.  For instance, when a button is clicked using a pointing device, the focus is generally not visually indicated, but when a text box needing user input has focus, focus is indicated. While focus styles are always required when users are navigating the page with the keyboard or when focus is managed via scripts, focus styles are not required when the user knows where they are putting focus, such as when they use a pointing device such as a mouse or finger to physically set focus on an element, unless that element continues to need user attention.

The `:focus` pseudo-class always matches the currently-focused element. The `:focus-visible` pseudo-class also matches the focused element, but only if the user needs to be informed where the focus currently is. Because the `:focus-visible` pseudo-class matches the focused element when needed, using the `:focus-visible` (instead of the `:focus` pseudo-class) allows authors to change the appearance of the focus indicator without changing when the focus indicator appears.
```

File: files/en-us/web/css/_colon_focus-visible/index.md [hunk 3]
```diff
@@ -21,6 +21,16 @@ This selector is useful to provide a different focus indicator based on the user
...
+
+Browser no longer visibly indicate focus (such as by drawing a “focus ring”), around each element when it has focus. Instead, they use a variety of heuristics to provide focus indicators only when it would be most helpful to the user.  For instance, when a button is clicked using a pointing device, the focus is generally not visually indicated, but when a text box needing user input has focus, focus is indicated. While focus styles are always required when users are navigating the page with the keyboard or when focus is managed via scripts, focus styles are not required when the user knows where they are putting focus, such as when they use a pointing device such as a mouse or finger to physically set focus on an element, unless that element continues to need user attention.
```
Comment (OnkarRuikar) [inline files/en-us/web/css/_colon_focus-visible/index.md:28]:
```suggestion
Browser no longer visibly indicate focus (such as by drawing a "focus ring"), around each element when it has focus. Instead, they use a variety of heuristics to provide focus indicators only when it would be most helpful to the user. For instance, when a button is clicked using a pointing device, the focus is generally not visually indicated, but when a text box needing user input has focus, focus is indicated. While focus styles are always required when users are navigating the page with the keyboard or when focus is managed via scripts, focus styles are not required when the user knows where they are putting focus, such as when they use a pointing device such as a mouse or finger to physically set focus on an element, unless that element continues to need user attention.
```

File: files/en-us/web/css/_colon_focus-visible/index.md [hunk 4]
```diff
@@ -23,7 +23,11 @@ This selector is useful to provide a different focus indicator based on the user
...
```
Comment (estelle) [inline files/en-us/web/css/_colon_focus-visible/index.md]:
```suggestion
Browser no longer visibly indicate focus (such as by drawing a "focus ring"), around each element when it has focus. Instead, they use a variety of heuristics to provide focus indicators only when it would be most helpful to the user. For instance, when a button is clicked using a pointing device, the focus is generally not visually indicated, but when a text box needing user input has focus, focus is indicated. While focus styles are always required when users are navigating the page with the keyboard or when focus is managed via scripts, focus styles are not required when the user knows where they are putting focus, such as when they use a pointing device such as a mouse or finger to physically set focus on an element, unless that element continues to need user attention.
```

**Review by estelle — APPROVED** [review]
Thanks! 🎉

**Comment by SteveJonk** [comment]
So much clearer, thanks!
